feat: add roster-bootstrap-rsa plugin design doc and scaffold for WRB streaming (#2631)
hiero-ledger/hiero-block-node: block-node/roster-bootstrap-rsa/README.md

# RSA Roster Bootstrap Plugin

**Module:** `org.hiero.block.node.roster.bootstrap.rsa`
**Plugin class:** `org.hiero.block.node.roster.bootstrap.rsa.RsaRosterBootstrapPlugin`
**Design doc:** [`docs/design/wrb-streaming/bootstrap-roster-plugin.md`](../../docs/design/wrb-streaming/bootstrap-roster-plugin.md)
**Implements:** [#2561](https://github.com/hiero-ledger/hiero-block-node/issues/2561)
**Part of epic:** [#2509](https://github.com/hiero-ledger/hiero-block-node/issues/2509) — Phase 2a WRB Streaming

---

## Purpose

This plugin loads the consensus node roster — a mapping of `node_id → RSA public key` — at Block
Node startup. The loaded roster is made available to all plugins ensuring the verification plugin
has the details to verify Wrapped Record Blocks (WRBs) carrying `SignedRecordFileProof` block proofs.

WRBs are produced by Consensus Nodes during Phase 2a of the Hiero network upgrade. They carry a
set of gossiped RSA signatures from every node in the current roster as their block proof. Without
a populated roster the Block Node cannot accept any WRB, so the plugin fails startup fast when
the roster cannot be loaded.

---

## How it works

> 1. **File-first:** On startup the plugin checks for a local bootstrap file at the configured path
>    (default `data/config/rsa-bootstrap-roster.pb`). If found, the roster is loaded from that file.

Reviewer: ```suggestion
1. **File-first:** On startup the application state facility checks for a local bootstrap file at the configured path
   (default `data/config/rsa-bootstrap-roster.pb`). If found, the roster is loaded from that file.
```

2. **Mirror Node fallback:** If no local file is present, the roster is fetched from the Hedera
   Mirror Node REST API (`GET /api/v1/network/nodes`, paginated). The result is written to the
   bootstrap file for future restarts.

> 3. **Fail fast:** If neither source succeeds startup is aborted with a clear error message.

Reviewer: We might want block node detail status as an additional source.

4. **No runtime reload:** The roster is loaded once and does not change for the lifetime of the
   BN instance. An address-book change requires a restart with a refreshed bootstrap file.

---

## Bootstrap file format

> The bootstrap file is a **binary protobuf serialization** of the `NodeAddressBook` message from
> `basic_types.proto` (`hiero-consensus-node`). This reuses the on-chain address book format stored
> at Hedera file `0.0.102`, avoiding a bespoke schema.

Reviewer: ```suggestion
The bootstrap file is a **binary protobuf serialization** of the `NodeAddressBook` message from
`basic_types.proto` (`hiero-consensus-node`). This reuses the on-chain address book format used for
Hedera file `0.0.102`, avoiding a bespoke schema.
```

|-----------------------------------------------------|------------------------------------------------|-----------------------------------------------------------|
| `roster.bootstrap.enabled`                          | `true`                                         | Enable/disable the plugin.                                |
| `roster.bootstrap.filePath`                         | `data/config/rsa-bootstrap-rosterp.pb`         | Path to the local bootstrap file (binary protobuf).       |
| `roster.bootstrap.mirrorNode.baseUrl`               | `https://testnet-public.mirrornode.hedera.com` | Mirror Node base URL.                                     |
| `roster.bootstrap.mirrorNode.connectTimeoutSeconds` | `5`                                            | Connect timeout for Mirror Node calls.                    |
| `roster.bootstrap.mirrorNode.readTimeoutSeconds`    | `10`                                           | Read timeout for Mirror Node calls.                       |
| `roster.bootstrap.mirrorNode.pageSize`              | `100`                                          | Nodes per page for paginated Mirror Node calls (max 100). |

---

> ## Relationship to `TssBootstrapPlugin`

Reviewer 2: `TssBootstrapPlugin` has been renamed RosterBootstrapTssPlugin

This plugin parallels `TssBootstrapPlugin` in structure. Both plugins:

> - Implement `BlockNodePlugin` and perform all work in `init()`.
> - Use a binary protobuf bootstrap file in `data/config/`.
> - Populate a field on `BlockNodeContext` for downstream consumers.

Reviewer: ```suggestion
- Implement `BlockNodePlugin` and perform all work in `init()`.
- Use a binary protobuf bootstrap file in `data/config/` managed by Application State Facility.
- Populate a field on `BlockNodeContext` for downstream consumers.
```
